infrastructure/v1/servicemesh_types.go

```diff
@@ -38,3 +41,30 @@ type IngressGatewaySpec struct {
 	// the for Ingress Gateway.
 	Certificate CertificateSpec `json:"certificate,omitempty"`
 }
+
+type AuthSpec struct {
+	// Namespace where it is deployed.
+	// +kubebuilder:default=auth-provider
+	Namespace string `json:"namespace,omitempty"`
+	// Authorino holds configuration of Authorino service used as external authorization provider.
+	Authorino AuthorinoSpec `json:"authorino,omitempty"`
+}
+
+type AuthorinoSpec struct {
+	// Name specifies how external authorization provider should be called.
+	// +kubebuilder:default=authorino-mesh-authz-provider
+	Name string `json:"name,omitempty"`
+	// Audiences is a list of the identifiers that the resource server presented
+	// with the token identifies as. Audience-aware token authenticators will verify
+	// that the token was intended for at least one of the audiences in this list.
+	// If no audiences are provided, the audience will default to the audience of the
+	// Kubernetes apiserver (kubernetes.default.svc).
+	// +kubebuilder:default={"https://kubernetes.default.svc"}
+	Audiences []string `json:"audiences,omitempty"`
+	// Label narrows amount of AuthConfigs to process by Authorino service.
+	// +kubebuilder:default=authorino/topic=odh
+	Label string `json:"label,omitempty"`
+	// Image allows to define a custom container image to be used when deploying Authorino's instance.
+	// +kubebuilder:default="quay.io/kuadrant/authorino:v0.16.0"
+	Image string `json:"image,omitempty"`
```

Comment:
These are internal things, based on initial feedback from #605 we should not expose it directly in this API

Reply:
I was going to add and ENV flag to disable by default this.

What I've learnt in the past is that these settings are useful not only for dev, but also for people who deploy a private registry where images are mirrored.

So, I agree the operator should provide non-customizable image on a standard install. Yet, it is still quite useful to provide a way to set your own images (not only for development). But the intent must be explicit.